lib/grp.gi

```diff
@@ -5134,6 +5132,9 @@ local  i, U, gens,test;
   return gens;
 end);
 
+InstallMethod(SmallGeneratingSet,"generators subset",
+  [IsGroup and HasGeneratorsOfGroup],SMALLGENERATINGSETGENERIC);
```

Comment:
What's the motivation for this change? Do you plan to use `SMALLGENERATINGSETGENERIC` directly elsewhere?

Reply:
Yes. (So far privately, but it might at some point become library code.)